File: YEAR 2/SEM 1/Advanced Programming Methods/LAB 4/src/Util/Dictionary/MyIDictionary.java
package Util.Dictionary;

import java.util.Collection;
import java.util.Map;
import java.util.Set;

public interface MyIDictionary<K,V> {
    void put(K key,V value);
    boolean containsKey(Object key);
    boolean containsValue(Object value);
    V get(Object key);
    V remove(K key);
    Collection<V> values();
    boolean isEmpty();
    int size();
    Set<Map.Entry<K,V>> entrySet();
    void clear();
}

File: YEAR 2/SEM 1/Advanced Programming Methods/ToyLanguage/ToyLanguage/Util/Dictionary/MyIDictionary.cs
using System.Collections.Generic;

namespace ToyLanguage.Util.Dictionary
{
    interface MyIDictionary<TKey,TValue>
    {
        void Add(TKey key, TValue value);
        bool Remove(TKey key);
        int Count
        {
            get;
        }

        bool ContainsKey(TKey key);

        void Update(TKey key,TValue value);

        bool TryGetValue(TKey key, out TValue value);

        Dictionary<TKey, TValue>.Enumerator GetEnumerator();
    }
}
